src/renderer/components/+workloads/workloads.route.ts

```diff
@@ -35,6 +28,19 @@ export const cronJobsRoute: RouteProps = {
   path: "/cronjobs"
 };
 
+export const workloadsRoute: RouteProps = {
```

Comment:
Are these refactorings necessary for the fix?

Reply:
Yes, they're tightly coupled. Previous `workloadsRoute` (and similar) used following sentence:
```
return Workloads.tabRoutes.map(({ routePath }) => routePath).flat();
```
`Workloads.tabRoutes` checks if some resource is allowed then pushed it to `routes` array
```
if (isAllowedResource("pods")) {
      routes.push({
        title: "Pods",
        component: Pods,
        url: podsURL({ query }),
        routePath: podsRoute.path.toString()
      });
    }
```
The key here that `isAllowedResouse` is using observable `getHostedCluster` inside. This causes an `<App/>` to rerender. So, by removing references to observable field we eliminate excessive rerenderings.